types/Img.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace plugin\struct\types;
+
+use plugin\struct\meta\ValidationException;
+
+class Img extends AbstractBaseType {
+
+    protected $config = array(
+    );
+
+    /**
+     * Output the stored data
+     *
+     * If outputted in an aggregation we collect the images into a gallery.
+     *
+     * @param string|int $value the value stored in the database
+     * @param \Doku_Renderer $R the renderer currently used to render the data
+     * @param string $mode The mode the output is rendered in (eg. XHTML)
+     * @return bool true if $mode could be satisfied
+     */
+    public function renderValue($value, \Doku_Renderer $R, $mode) {
+        if (empty($value)) {
+            return false;
+        }
+
+        $returnLink = null;
+        if ($mode == 'xhtml') {
+            $returnLink = true;
+        }
+        if (strpos($value, '://') === false) {
+            $html = $R->internalmedia($value, null, null, null, null, null, 'direct', $returnLink);
+        } else {
+            $html = $R->externalmedia($value, null, null, null, null, null, 'direct', $returnLink);
+        }
+
+        if ($mode == 'xhtml') {
+            $hash = '';
+            if (strrpos($R->doc,'</table>') < strrpos($R->doc,'class="table structaggregation')) {
+                $hash = substr($R->doc,strpos($R->doc,'hash:',strrpos($R->doc,'class="table structaggregation'))+5,32);
+                $hash = "[gal-$hash]";
+            }
```

Comment:
Ah, now I understand what you meant. There's an easier way. The renderer object you are passed here is the same object you used earlier in startScope. You could simply store the hash in $renderer->info['struct_table_hash'] in startScope and remove it again in stopScope. Later check here, if the value is set and act accordingly.